### DISCUSSION TITLE
Launch multiple simulation with  a couple of variable specific for each simulation.

### DISCUSSION THREAD

**Discussion by rodolphebauer** [gama-platform/gama#346] [Q&A]
Hello everyone,

I'm a user of Gama, more specifically of the MAELIA plateform. To run my simulations, I use the batch fonction of gama. However, for each simulation I run, I want to change two different parameters in my launcher. I know how to do with one, by using the function "among" and giving to gama each value of the parameter needed, but if I use "among" for two parameters, gama performs a simulation for each combinaison of the values for the first and the second parameter. I want gama to run the simulation with the first value of both parmater, then the second etc...

Let's take an exemple : 
parameter 'name : '     var: name     <- 'titi' 
							among: ['titi', 'toto', 'tata']
parameter 'color : '     var: name     <- 'blue' 
							among: ['blue', 'green', 'yellow']
I want to have three simulations with the values ['titi','blue'], ['toto', 'green'], ['tata', 'yellow'].

Is there a way to do that ? 

Many thanks !

**Comment by lesquoyb** [comment]
Hello, one solution I can think of is this:
```
model tupleofparameters

global {
	map<string,int> varMap <- map("var1"::1, "var2"::2);
	int var1;
	int var2;
	
	init {
		var1 <- varMap["var1"];
		var2 <- varMap["var2"];
	}
}

experiment test type:batch until:cycle=1{
	
	parameter "Variable tuples" var:varMap among:[	map("var1"::1, "var2"::2), 
													map("var1"::2, "var2"::3), 
													map("var1"::3, "var2"::4), 
													map("var1"::4, "var2"::5)
	];
	
	reflex print{
		write "======";
		write sample(var1);
		write sample(var2);
	}
	
}
```
The idea is to use a map of values as your only parameter instead of multiple individual parameters.
Then in the init part of the model you use that map to initialise all your individual parameters.
Not the most elegant but it works

**Reply by rodolphebauer** [reply]
Thanks for the help !

It seems to work but for a reason I don't understand the simulation is slowed significantly when I applied this method. Furthemore, the simulation doesn't end so the batch technically doesn't work. We're trying to figure why but we didn't found an answer yet. 

thanks anyway.

**Reply by lesquoyb** [reply]
Both the fact that it's significantly slower and that the simulation doesn't stop are absolutely unexpected!
If you can isolate it in a (somewhat) smaller model could you share it so I can have a look ?
